### DISCUSSION TITLE
how to profile dynamic endpoint while using skywalking?

### DISCUSSION THREAD

**Discussion by jmbkeyes** [apache/skywalking#8402] [Q&A]
I'm new to skywalking. I came across issue on profiling the dynamic endpoint.
**dynamic endpoint:**  a restful api of micro-service, which contains dynamic parameters in the url(e.g.  http://xxx.com/api/test/hello/{name}, {name} is dynamical parameter, its value is gotten from url).
I can profile the static endpoint(e.g. http://xxx.com/api/test/info), but can't profile the task on dynamic endpoint. Could someone help me?
![1641887566(1)](https://user-images.githubusercontent.com/12583734/148902424-d56575c9-d292-4d09-a202-9bcc9286339b.png)
![1641887588(1)](https://user-images.githubusercontent.com/12583734/148902463-627b3ebc-4773-45a4-ac70-1b5904e0be9c.png)

**Comment by wu-sheng** [comment]
It is not supported. Endpoint is endpoint, you could parameterize it or use raw URI according to your codes and plugins.
But you can't ask for grouped statistics, and profiling for specific one at the same time. This doesn't make sense.

**Reply by jmbkeyes** [reply]
Firstly, thanks for your reply. 
1. parameterize it  
    Does it mean that parameterize the variable name?   (e.g.  http://xxx.com/api/test/hello/abc)
2. use raw URI according to your codes and plugins
    Not sure how to do it?

3. As you know, restful api sometimes put parameter in the endpoint name(e.g. /order/detail/1， /order/detail/2), how to profile in such case.

**Reply by wu-sheng** [reply]
In your (3) case, if you put endpoint name as `/oder/details/{id}` due to SpringMVC plugin, then, profiling is randomly picking a request according to latency.
(1), don't understand your question. Also, your example doesn't make things clear too.
(2), it depends on which plugins you are using. There are many plugins from different contributors globally. I don't have a simple answer.

**Reply by jmbkeyes** [reply]
For the case 3, I add a test restful api(it can be access by http://10.0.0.5:9094/api/test/{id}) in microservice
![image](https://user-images.githubusercontent.com/12583734/149074385-ec04428d-1078-4d7a-b5f0-c6eaa12361a3.png)
Then I start the microservice,  and I can see the the service in skywalking ui after calling the api.
![1641968394(1)](https://user-images.githubusercontent.com/12583734/149074559-22ce048a-e14d-429a-b7aa-9a8d992fdc78.png)

And then I add a profile task with endpoint name as /api/test/{id}
![1641968616(1)](https://user-images.githubusercontent.com/12583734/149075099-7867bb68-3e34-40b0-81b5-eb2ceec0f3f7.png)
Finally I call the rest api more than 5 times, but there're no profile data.
![1641968733(1)](https://user-images.githubusercontent.com/12583734/149075256-1e0beb73-cc0f-4468-ba44-863a4ea4db41.png)

**Reply by wu-sheng** [reply]
I think I should not debug for you. This is not how open source works. UI and env are at your side and totally private.
